### DISCUSSION TITLE
Setting proxy on Appium for MacOS app

### DISCUSSION THREAD

**Discussion by MagdelineNg** [appium/appium#19563] [Q&A]
I am using [Appium Python Client](https://github.com/appium/python-client) + [Mac2 driver ](https://github.com/appium/appium-mac2-driver) to automate my MacOS Desktop app, and I want to configure the driver to a http and https proxy. How do I pass it in as options?

```
from appium import webdriver as appium_webdriver
from appium.options.mac import Mac2Options

options = Mac2Options()
options.bundle_id = BUNDLE_ID
options.proxy = {
    "proxyType": "manual",
    "httpProxy": PROXY_DOMAIN,
    "httpProxyPort": PROXY_PORT,
    "httpPassword": PROXY_PASSWORD,
    "httpUser": PROXY_USER,
}
driver = appium_webdriver.Remote(f"http://{APPIUM_HOST}:{APPIUM_PORT}", options=options)
```

I am receiving an error: selenium.common.exceptions.InvalidArgumentException: Message: Only Proxy objects can be passed in.

Next, I tried setting the proxy using Python's subprocess module before starting the driver. It worked as I am receiving alerts shown below to use my proxy's information after I start the driver. 
<img width="456" alt="Screenshot 2023-12-21 at 11 21 58 AM" src="https://github.com/appium/appium/assets/85625305/2554f16f-68e1-4805-88ae-ef4410b4f808">

<img width="432" alt="Screenshot 2023-12-21 at 11 22 51 AM" src="https://github.com/appium/appium/assets/85625305/54658b6b-feff-472c-ba71-f1cf4422305c">

The code for the above:
```
def set_proxy():
    subprocess.run(['networksetup', '-setwebproxy', 'Wi-fi', '10.49.85.215', '443', 'on', 'username', 'pw'])
    subprocess.run(['networksetup', '-setwebproxystate', 'Wi-fi', 'on'])

    subprocess.run(['networksetup', '-setsecurewebproxy', 'Wi-fi', '10.49.85.215', '443', 'on', 'username', 'pw'])
    subprocess.run(['networksetup', '-setsecurewebproxystate', 'Wi-fi', 'on'])

def check_proxy():
    web_proxy_status = subprocess.run(['networksetup', '-getwebproxy', 'Wi-fi'], capture_output=True, text=True)
    if 'Enabled: Yes' in web_proxy_status.stdout:
        print('Web proxy is enabled')
    else:
        print('Web proxy is not enabled')

    secure_web_proxy_status = subprocess.run(['networksetup', '-getsecurewebproxy', 'Wi-fi'], capture_output=True, text=True)
    if 'Enabled: Yes' in secure_web_proxy_status.stdout:
        print('Secure web proxy is enabled')
    else:
        print('Secure web proxy is not enabled')
 
set_proxy()
check_proxy()
driver = appium_webdriver.Remote(f"http://{APPIUM_HOST}:{APPIUM_PORT}", options=options)
```
      
        
How do I ensure that the Appium driver is granted permissions so these prompts do not appear?

**Comment by jlipps** [comment]
```
selenium.common.exceptions.InvalidArgumentException: Message: Only Proxy objects can be passed in
```

Did you verify that the form of the capabilities you are using is correct? It seems maybe your client doesn't like it.

**Reply by MagdelineNg** [reply]
I'm using appium python client and setting proxy as an attribute in options. There is some declaration(docstring) shown when I hover, so I assume it does support the options.proxy attribute. 
<img width="425" alt="image" src="https://github.com/appium/appium/assets/85625305/5b418b9c-03cd-4f1c-927d-c3464e131158">
